docs/plugins.md

```diff
@@ -1,87 +1,86 @@
+To create your own BitOps, you will need two files;
+* **[bitops.config.yaml](../bitops.config.yaml)**: Contains configuration attributes that will modify how BitOps behaves
+* **[Dockerfile](../prebuilt-config/dockerfile.template)**: Needs to use the BitOps base image in the `FROM` directive
 
-## Local plugins not from GitHub
 
-If users want to use BitOps with arbitrary plugins, then they need to create their own repo that looks something like:
-```
-my-bitops
-├── Dockerfile
-└── plugin.config.yaml
+### bitops.config.yaml
+Best explained with an example, The default `bitops.config.yaml` looks like this:
 ```
+bitops:
+  # The `bitops.config.yaml` file contains the configuration values for the BitOps core.
+  #   - Changing values will require that a new image be built
+  #   - Customize your BitOps image by modifying the values found in the `bitops.config.yaml`
 
-The Dockerfile looks like:
-```
-FROM bitops-core
-COPY plugin.config.yaml .
-RUN python scripts/setup/install_plugins.py
-```
+  fail_fast: true     # When set, will exit if any warning+ is raised, otherwise only exit on critical error
+  # LEVELS: [ DEBUG, INFO, WARNING, ERROR, CRITICAL ]
+  logging:      
+    level: DEBUG              # Sets the logging level
+    color:
+      enabled: true           # Enables colored logs
+    filename: bitops-run      # log filename
+    err: bitops.logs          # error logs filename
+    path: /var/logs/bitops    # path to log folder
+  opsrepo_root_default_dir: _default
+  plugins:    
+    aws:
+      source: https://github.com/bitops-plugins/aws
+    terraform:
+      source: https://github.com/bitops-plugins/terraform
+    cloudformation:
+      source: https://github.com/bitops-plugins/cloudformation
+    helm:
+      source: https://github.com/bitops-plugins/helm
+    kubectl:
+      source: https://github.com/bitops-plugins/kubectl
+    ansible:
+      source: https://github.com/bitops-plugins/ansible
+  deployments:
+    cloudformation:
+      plugin: cloudformation
+    terraform:
+      plugin: terraform
+    helm:
+      plugin: helm
+    ansible:
+      plugin: ansible
 
-To add your own local plugins the my-bitops repo should be extended to contain my "local" plugins
-```
-my-bitops
-├── Dockerfile
-├── plugin.config.yaml
-└── plugins
-    └── my-custom-plugin-1
 ```
-These local plugins need to be passed in to the docker context by include them in the Dockerfile
+The repo for each plugin must be a `git clone`-able url. The name can be anything.
+
+The order that plugins run is dependent on the `deployments` section. If a `depoyments` section isn't provided, it will attempt to process all folders in the `BITOPS_ENVIRONMENT` directory in alphabetical order.
+
+**Dockerfile**
+The only content that is needed to create a custom image is;
+
 ```
-FROM bitops-core
-COPY plugin.config.yaml .
-ADD ./plugins/ /scripts/plugins/
-RUN python scripts/setup/install_plugins.py
+FROM bitovi/bitops:latest-base
 ```
 
```

Comment:
Even if the final step is as simple as `docker build .`, still worth to include the command instruction for copy-paste here.